File: src/Servers/PresenceConnectionManager/src/Entity/Structure/Misc/SDKRevision.cs
using UniSpy.Server.PresenceConnectionManager.Entity.Enumerate;

namespace UniSpy.Server.PresenceConnectionManager.Entity.Structure.Misc
{
    public sealed class SdkRevision
    {
        public SdkRevisionType? SDKRevisionType { get; set; }
        public bool IsSDKRevisionValid => SDKRevisionType == 0 ? false : true;
        public bool IsSupportGPINewAuthNotification => (SDKRevisionType ^ Enumerate.SdkRevisionType.GPINewAuthNotification) != 0 ? true : false;
        public bool IsSupportGPINewRevokeNotification => (SDKRevisionType ^ Enumerate.SdkRevisionType.GPINewRevokeNotification) != 0 ? true : false;
        public bool IsSupportGPINewStatusNotification => (SDKRevisionType ^ Enumerate.SdkRevisionType.GPINewStatusNotification) != 0 ? true : false;
        public bool IsSupportGPINewListRetrevalOnLogin => (SDKRevisionType ^ Enumerate.SdkRevisionType.GPINewListRetrevalOnLogin) != 0 ? true : false;
        public bool IsSupportGPIRemoteAuthIDSNotification => (SDKRevisionType ^ Enumerate.SdkRevisionType.GPIRemoteAuthIDSNotification) != 0 ? true : false;
        public bool IsSupportGPINewCDKeyRegistration => (SDKRevisionType ^ Enumerate.SdkRevisionType.GPINewCDKeyRegistration) != 0 ? true : false;
        public SdkRevision()
        {
        }
    }
}

File: src/frontends/gamespy/protocols/presence_connection_manager/aggregates/sdk_revision.py
from frontends.gamespy.protocols.presence_connection_manager.aggregates.enums import SdkRevisionType


class SdkRevision:
    def __init__(self, sdk_type: SdkRevisionType):
        assert isinstance(sdk_type, SdkRevisionType)
        self.sdk_type = sdk_type

    @property
    def is_sdk_revision_valid(self):
        return False if self.sdk_type == 0 else True

    @property
    def is_support_gpi_new_auth_notification(self):
        return (
            True
            if (self.sdk_type ^ SdkRevisionType.GPINEW_AUTH_NOTIFICATION) != 0
            else False
        )

    @property
    def is_support_gpi_new_revoke_notification(self):
        return (
            True
            if (self.sdk_type ^ SdkRevisionType.GPINEW_REVOKE_NOTIFICATION) != 0
            else False
        )

    @property
    def is_support_gpi_new_status_notification(self):
        return (
            True
            if (self.sdk_type ^ SdkRevisionType.GPINEW_STATUS_NOTIFICATION) != 0
            else False
        )

    @property
    def is_support_gpi_new_list_retreval_on_login(self):
        return (
            True
            if (self.sdk_type ^ SdkRevisionType.GPINEW_LIST_RETRIEVAL_ON_LOGIN) != 0
            else False
        )

    @property
    def is_support_gpi_remote_auth_ids_notification(self):
        return (
            True
            if (self.sdk_type ^ SdkRevisionType.GPIREMOTE_AUTH_IDS_NOTIFICATION) != 0
            else False
        )

    @property
    def is_support_gpi_new_cdkey_registration(self):
        return (
            True
            if (self.sdk_type ^ SdkRevisionType.GPINEW_CD_KEY_REGISTRATION) != 0
            else False
        )
